### ISSUE TITLE
Failed to connect to Existing DOJO

### ISSUE THREAD

**Issue by bidelakbari** [Samourai-Wallet/whirlpool-gui#10]
I tried to pair whirlpool-gui to an existing dojo but no luck on that
- already paired the samourai-wallet with the dojo
- can access the dojo maintenance page (thorough Tor ofc)
- whirlpool-gui's "standalone GUI" option works and pairs with the samourai wallet
 
but in <code>Connect to existing DOJO/Nodl/CLI</code> option I have entered the v3 onion address and my api-key but I get the following error:
<code>Connection failed: could not connect to CLI (may take a few seconds to start...)</code>

What I'm missing here?

**Comment by zeroleak** [comment]
Hello, 
The "Connect to existing DOJO/Nodl/CLI" feature is for running Whirlpool from a CLI (command-line script with --listen argument) on a remote system, and remote-control it from the GUI.
If you wish to pair Whirlpool with your own Dojo backend, this will be available in the next GUI release (very soon).

**Comment by bidelakbari** [comment]
Hello and thank you for the clarification

> If you wish to pair Whirlpool with your own Dojo backend, this will be available in the next GUI release (very soon).

nice, really looking forward to it then.

I suppose currently pairing to the wallet is needed for getting the xpub and xpriv which is required for participating in the mixing round; **if my assumption is correct**, is it correct to say:

_When the second option<code>(Connect to existing DOJO/Nodl/CLI)</code> is available there would be no need to pair with the wallet and the gui will connect to the dojo server directly to get required information to participate in mixing rounds._ 

One last question:
Privacy wise what is the difference between pairing with the wallet and connecting to the dojo server directly (coming feature) for whirlpool mixing?

/thanks

**Comment by zeroleak** [comment]
hi,
Sorry for the late reply.
- The second option `_Connect to existing DOJO/Nodl/CLI_` was renamed to  `_Connect to remote CLI_`. It enables you to keep mixing while GUI is closed, by running CLI on a remote server. It has no effect on privacy.
- Dojo pairing is now available in 0.8.0. Just pair Whirlpool with an already Dojo-paired SamouraiWallet, it will be automatically detected by Whirlpool. Dojo avoids using Samourai backend for utxos listing from xpub, for better privacy.

To sum up, you can have:
- standalone GUI without Dojo
- standalone GUI with Dojo
- remote CLI without Dojo
- remote CLI with Dojo

and still need to pair with SamouraiWallet, even when using Dojo, because Dojo can index multiple wallet :-)
